pkgs/applications/networking/mullvad-vpn/default.nix

```diff
@@ -1,145 +1,81 @@
-{ stdenv
-, lib
-, fetchurl
-, dpkg
-, alsa-lib
-, atk
-, cairo
-, cups
-, dbus
-, expat
-, fontconfig
-, freetype
-, gdk-pixbuf
-, glib
-, pango
-, nspr
-, nss
-, gtk3
-, mesa
-, libGL
-, wayland
-, xorg
-, autoPatchelfHook
-, systemd
-, libnotify
-, libappindicator
-, makeWrapper
-, coreutils
-, gnugrep
+{
+  lib,
+  buildNpmPackage,
+  protobuf,
+  protoc-gen-js,
+  grpc-tools,
+  makeWrapper,
+  electron,
+  mullvad,
 }:
 
-let
-  deps = [
-    alsa-lib
-    atk
-    cairo
-    cups
-    dbus
-    expat
-    fontconfig
-    freetype
-    gdk-pixbuf
-    glib
-    pango
-    gtk3
-    libappindicator
-    libnotify
-    mesa
-    xorg.libX11
-    xorg.libXScrnSaver
-    xorg.libXcomposite
-    xorg.libXcursor
-    xorg.libXdamage
-    xorg.libXext
-    xorg.libXfixes
-    xorg.libXi
-    xorg.libXrandr
-    xorg.libXrender
-    xorg.libXtst
-    xorg.libxcb
-    xorg.libxshmfence
-    nspr
-    nss
-    systemd
-  ];
+buildNpmPackage rec {
+  pname = "mullvad-vpn";
+  inherit (mullvad) src version;
 
-  version = "2024.4";
+  patches = [ ./electron-builder.patch ];
 
-  selectSystem = attrs: attrs.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  postPatch = ''
+    substituteInPlace gui/tasks/distribution.js \
+        --replace-fail 'function productVersion(extra_args) {' 'function productVersion(extra_args) { return "${version}";' \
+        --subst-var-by electron_dist ${electron.dist} \
+        --subst-var-by electron_version ${electron.version}
+  '';
```

Comment:
```suggestion
```
As you're already patching `distribution.js` in a `.patch` file, I'd recommend copying the `substituteInPlace` changes to that patch file for consistency.

Reply:
I don't hate that we have both since it automates for electron version updates, which could move independently, kindof